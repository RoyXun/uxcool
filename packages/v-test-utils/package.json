{
  "name": "@suning/v-test-utils",
  "version": "1.0.10",
  "main": "src/index.js",
  "module": "src/index.js",
  "license": "MIT",
  "scripts": {
    "start": "npm run server",
    "compile": "v-tools run compile",
    "server": "v-tools run server"
  },
  "peerDependencies": {
    "vue": ">= 2.5.*"
  },
  "devDependencies": {
    "@suning/v-tools": "^1.0.35",
    "@vue/test-utils": "^1.0.0-beta.16"
  },
  "dependencies": {
    "@suning/v-trigger": "^1.0.64",
    "@vue/server-test-utils": "^1.0.0-beta.30",
    "jquery": "^3.3.1",
    "vue-server-renderer": "^2.6.11"
  }
}
