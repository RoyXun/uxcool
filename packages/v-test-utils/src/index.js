import Vue from 'vue';

import $ from 'jquery';
import { mount, createWrapper } from '@vue/test-utils';
import Trigger from '@suning/v-trigger';

export * from '@vue/test-utils';

// export * as server from '@vue/server-test-utils';

export { $, $ as jQuery };
// ms
export async function waitTime(delay = 20) {
  await new Promise((resolve) => {
    setTimeout(() => {
      resolve();
    }, delay);
  });
}

export function removePopup() {
  document.body.innerHTML = '';
}

const vueInst = new Vue();
export function getVueCreateElement() {
  return vueInst.$createElement;
}

/**
 * 获取Trigger实例
 * @param {Wrapper} wrapper
 */
export async function getPortal(wrapper) {
  const trigger = wrapper.find(Trigger);
  const portal = createWrapper(trigger.vm.portal.$el);
  return portal;
}

/**
 * 触发事件
 * @param {Wrapper} wrapper
 * @param {string} eventName
 * @param {number} delay
 */
export async function triggerEvent(wrapper, eventName, delay = 20) {
  wrapper.trigger(eventName);
  await waitTime(delay);
}

export function mountPickerFactory(Component) {
  return async (opts = {}) => {
    const wrapper = mount(Component, {
      ...opts,
    });
    await waitTime();
    return wrapper;
  };
}

export function destroyWrapper(wrapper) {
  wrapper.destroy();
}
