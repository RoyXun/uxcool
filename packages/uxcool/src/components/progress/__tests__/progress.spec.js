import { mountPickerFactory, waitTime, getVueCreateElement } from '@suning/v-test-utils';
import Progress from '..';

const mountProgress = mountPickerFactory(Progress);
describe('progress', () => {
  it('render default progress correctly', async () => {
    const wrapper = await mountProgress();
    expect(wrapper.html()).toMatchSnapshot();

    wrapper.setProps({
      percentage: 40,
      strokeWidth: 10,
      strokeLinecap: 'square',
      strokeColor: 'skyblue',
    });
    await waitTime();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('render custom progress correctly', async () => {
    // eslint-disable-next-line
    const h = getVueCreateElement();
    const wrapper = await mountProgress({
      propsData: {
        percentage: 40,
      },
    });
    expect(wrapper.find('.ux-progress-text').text()).toBe('40%');
    expect(wrapper.contains('.ux-progress-text')).toBeTruthy();

    wrapper.setProps({ format: (percentage) => <span>{`${percentage}é¡¹`}</span> });
    await waitTime();
    expect(wrapper.html()).toMatchSnapshot();

    wrapper.setProps({ hideInfo: true });
    await waitTime();
    expect(wrapper.contains('.ux-progress-text')).not.toBeTruthy();
  });

  it('render three status progress correctly', async () => {
    const wrapper = await mountProgress();
    expect(wrapper.contains('.ux-progress-status-success')).not.toBeTruthy();

    wrapper.setProps({ status: 'success' });
    await waitTime();
    expect(wrapper.contains('.ux-progress-status-success')).toBeTruthy();
    expect(wrapper.contains('.ux-progress-status-exception')).not.toBeTruthy();

    wrapper.setProps({ status: 'error' });
    await waitTime();
    expect(wrapper.contains('.ux-progress-status-exception')).toBeTruthy();
    expect(wrapper.contains('.ux-progress-status-active')).not.toBeTruthy();

    wrapper.setProps({ status: 'active' });
    await waitTime();
    expect(wrapper.contains('.ux-progress-status-active')).toBeTruthy();
  });

  it('render circle progress correctly', async () => {
    const wrapper = await mountProgress({
      propsData: {
        type: 'circle',
        percentage: 30,
      },
    });
    expect(wrapper.find('.ux-progress-inner').element.style.width).toBe('120px');

    wrapper.setProps({ width: 80 });
    await waitTime();
    expect(wrapper.find('.ux-progress-inner').element.style.width).toBe('80px');

    wrapper.setProps({ gapDegree: 60, gapPosition: 'bottom' });
    await waitTime();
    // inner element is svg, may use snapshot to test
    expect(wrapper.element).toMatchSnapshot();
  });

  it('render dashboard progress correctly', async () => {
    const wrapper = await mountProgress({
      propsData: {
        type: 'dashboard',
        percentage: 30,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it.each(['left', 'right', 'bottom'])('render circle on differe pos', async (pos) => {
    const wrapper = await mountProgress({
      propsData: {
        type: 'circle',
        percentage: 30,
        gapPosition: pos,
        status: 'active'
      },
    });
    expect(wrapper.find('.ux-progress-inner').element.style.width).toBe('120px');
  });

  it.each([-10, 110])('render circle with  abnormal percentage', async (percentage) => {
    const wrapper = await mountProgress({
      propsData: {
        type: 'circle',
        percentage,
        status: 'default'
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
