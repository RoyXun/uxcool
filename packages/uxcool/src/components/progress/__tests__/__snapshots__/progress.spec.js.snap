// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`progress render circle progress correctly 1`] = `
<div
  class="ux-progress ux-progress-circle ux-progress-show-info ux-progress-status-normal ux-progress-default"
>
  <div
    class="ux-progress-inner"
    style="width: 80px; height: 80px; font-size: 18px;"
  >
    <svg
      class="ux-progress-circle"
      viewBox="0 0 100 100"
    >
      <path
        class="ux-progress-circle-trail"
        d="M 50 50 m 0,47
          a 47,47 0 1 1 0,-94
          a 47,47 0 1 1 0,94"
        fill="none"
        stroke-width="6"
        style="stroke: #f3f3f3; stroke-dasharray: 246.0914245312005px 295.3097094374406px; stroke-dashoffset: -24.609142453120043px; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s;"
      />
      <path
        class="ux-progress-circle-path"
        d="M 50 50 m 0,47
          a 47,47 0 1 1 0,-94
          a 47,47 0 1 1 0,94"
        fill="none"
        stroke-linecap="round"
        stroke-width="6"
        style="stroke-dasharray: 73.82742735936014px 295.3097094374406px; stroke-dashoffset: -24.609142453120043px; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s;"
      />
    </svg>
    <span
      class="ux-progress-text"
    >
      30%
    </span>
  </div>
</div>
`;

exports[`progress render custom progress correctly 1`] = `
<div class="ux-progress ux-progress-line ux-progress-show-info ux-progress-status-normal ux-progress-default">
  <div>
    <div class="ux-progress-outer">
      <div class="ux-progress-inner">
        <div class="ux-progress-bg" style="width: 40%; height: 8px; border-radius: 50px;"></div>
      </div>
    </div><span class="ux-progress-text"><span>40é¡¹</span></span>
  </div>
</div>
`;

exports[`progress render dashboard progress correctly 1`] = `
<div
  class="ux-progress ux-progress-circle ux-progress-show-info ux-progress-status-normal ux-progress-default"
>
  <div
    class="ux-progress-inner"
    style="width: 120px; height: 120px; font-size: 24px;"
  >
    <svg
      class="ux-progress-circle"
      viewBox="0 0 100 100"
    >
      <path
        class="ux-progress-circle-trail"
        d="M 50 50 m 0,47
          a 47,47 0 1 1 0,-94
          a 47,47 0 1 1 0,94"
        fill="none"
        stroke-width="6"
        style="stroke: #f3f3f3; stroke-dasharray: 233.78685330464046px 295.3097094374406px; stroke-dashoffset: -30.761428066400057px; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s;"
      />
      <path
        class="ux-progress-circle-path"
        d="M 50 50 m 0,47
          a 47,47 0 1 1 0,-94
          a 47,47 0 1 1 0,94"
        fill="none"
        stroke-linecap="round"
        stroke-width="6"
        style="stroke-dasharray: 70.13605599139214px 295.3097094374406px; stroke-dashoffset: -30.761428066400057px; transition: stroke-dashoffset .3s ease 0s, stroke-dasharray .3s ease 0s, stroke .3s, stroke-width .06s ease .3s;"
      />
    </svg>
    <span
      class="ux-progress-text"
    >
      30%
    </span>
  </div>
</div>
`;

exports[`progress render default progress correctly 1`] = `
<div class="ux-progress ux-progress-line ux-progress-show-info ux-progress-status-normal ux-progress-default">
  <div>
    <div class="ux-progress-outer">
      <div class="ux-progress-inner">
        <div class="ux-progress-bg" style="width: 0%; height: 8px; border-radius: 50px;"></div>
      </div>
    </div><span class="ux-progress-text">0%</span>
  </div>
</div>
`;

exports[`progress render default progress correctly 2`] = `
<div class="ux-progress ux-progress-line ux-progress-show-info ux-progress-status-normal ux-progress-default">
  <div>
    <div class="ux-progress-outer">
      <div class="ux-progress-inner">
        <div class="ux-progress-bg" style="width: 40%; height: 10px; border-radius: 0; background: skyblue;"></div>
      </div>
    </div><span class="ux-progress-text">40%</span>
  </div>
</div>
`;
